spec/acceptance/set_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'spec_helper_acceptance'
+require 'fileutils'
+
+describe 'pdk set' do
+  include_context 'with a fake TTY'
+
+  context 'when run outside of a module' do
+    describe command('pdk set') do
+      its(:exit_status) { is_expected.to eq 0 }
+      # Should show the command help
+      its(:stdout) { is_expected.to match(%r{pdk set \[subcommand\] \[options\]}) }
+      its(:stderr) { is_expected.to have_no_output }
+    end
```

Comment:
It'd be good to get some functional tests here as well

Reply:
The unit tests cover A LOT of those use cases.  So it felt strange to duplicate them

Reply:
Just a few cases to ensure the expected changes on disk which unit tests mock out. Set at system level, user level, project level, ensure the file gets created on disk at the appropriate path. Seems redundant but it covers cases like https://github.com/puppetlabs/pdk/pull/862

Reply:
Will do

Reply:
Actually, this will be VERY tricky to do in acceptance.  I'd either:

1. Have to mock File.write calls to the expected places on disk

OR

2. Make the config classes write to a temp file location

Point 1 is already done in `spec/unit/pdk/util_spec.rb` once #862 is merged

Point 2 is what's being used in this test suite already except only for user files.  If I was to use temporary `system` and `project` locations then it still wouldn't have picked up the issue #862

If I don't mock, then I'd be overwriting _actual_ file content which is not expected when running acceptance. If I want to write to _actual_ disk locations, this would be appropriate in the package-testing which is using VMs/Containters, not the localhost.

---

So yeah, that kind of testing needs to go in package-testing.

Reply:
@rodjek  I've added a new commit which adds some package-testing tests.